packages/strapi-parts/src/Layout/HeaderLayout.js

```diff
@@ -1,72 +1,131 @@
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import { H1, Subtitle, P, H2 } from '../Text';
 import { Box } from '../Box';
 import { Row } from '../Row';
 
-export const HeaderLayout = ({
-  navigationAction,
-  primaryAction,
-  secondaryAction,
-  subtitle,
-  title,
-  sticky,
-  ...props
-}) => {
-  if (sticky) {
+const useElementOnScreen = (options) => {
+  const containerRef = useRef(null);
+  const [isVisible, setIsVisible] = useState(true);
+
+  const cb = ([entry]) => {
+    setIsVisible(entry.isIntersecting);
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(cb, options);
+    if (containerRef.current) observer.observe(containerRef.current);
+
+    return () => {
+      if (containerRef.current) observer.unobserve(containerRef.current);
+    };
+  }, [containerRef, options]);
```

Comment:
We already have a `useIntersection` hook in `./helpers`, do you think it can be adjusted for this use case?

Reply:
Up on this one 😊